lib/postgrex/protocol.ex

```diff
@@ -1683,6 +1694,19 @@ defmodule Postgrex.Protocol do
     end
   end
 
+  defp maybe_disconnect({:error,
+         %Postgrex.Error{postgres: %{code: code}} = error,
+         %{disconnect_on_error_codes: codes} = state
```

Comment:
For roflscale reasons, let's have a clause before this one like this:

```elixir
    defp maybe_disconnect({:error, _, %{disconnect_on_error_codes: []}} = result), do: result
```

because that's the most common scenario after all.

Reply:
> For roflscale reasons

techempower here we go! oh wait, they don't do fault tolerance now do they?